lib/curl_get_line.c

```diff
@@ -41,17 +41,39 @@ char *Curl_get_line(char *buf, int len, FILE *input)
   bool partial = FALSE;
   while(1) {
     char *b = fgets(buf, len, input);
+
     if(b) {
       size_t rlen = strlen(b);
-      if(rlen && (b[rlen-1] == '\n')) {
+      if(!rlen)
+        break;
+
+      if(b[rlen-1] == '\n') {
+        /* b is \n terminated */
         if(partial) {
           partial = FALSE;
           continue;
         }
         return b;
       }
-      /* read a partial, discard the next piece that ends with newline */
-      partial = TRUE;
+      else if(feof(input)) {
+        if(partial) {
+          partial = FALSE;
+          continue;
+        }
+        if(rlen < (size_t) len) {
+          /* b is EOF terminated, insert missing \n */
+          b[rlen] = '\n';
+          return b;
+        }
+        else {
+          /* EOF terminated string using full buffer size
+           * This could either be truncated or correct.
+           * Most probably truncated.
+           */
+          partial = TRUE;
```

Comment:
As feof() already returned true here, surely this cannot be a truncated (partial) line?

Reply:
Indeed, it needs rephrasing. As it cannot be converted to valid by adding a '\n' within the given buffer size, my choice would be to ignore it like a partial line. 

We need to make a choice 
* Changing the fgets to a maximum of (len-1) would make it possible to add the \n, but it would also possibly introduce other backward compatibility issues (assuming someone uses a 4K config line somewhere)
* Accepting the last line without \n because it is not truncated would maybe introduce compatibility issues in other routines that call this function

My choice would be to ignore it (like in the code, but rephrased). It would allow all non-newline last lines under 4K long and ignore that one 4K long last line without a newline. 

```
else
  /* Max buffer reached + EOF - this line is impossible to add a \n to so we'll ignore it */
  break;
```

Agreed?

Reply:
I think that's totally fine to ignore in this case. It's just a special edge case anyway. Agreed!